pkg/net/rpc.go

```diff
@@ -278,3 +292,102 @@ func (c randChoice) Step(ctx context.Context) error {
 		Addrs:  c.h.Peerstore().Addrs(c.peer),
 	})
 }
+
+type gossip struct {
+	n     *neighborhood
+	sd    streamDialer
+	h     host.Host
+	peer  peer.ID
+	proto protocol.ID
+}
+
+func (g gossip) PushPull(ctx context.Context) ([]*peer.PeerRecord, error) {
+	s, err := g.sd.NewStream(ctx, g.peer, g.proto)
+	if err != nil {
+		return nil, err
+	}
+	defer s.Close()
+
+	if err = g.Push(ctx, s); err != nil {
```

Comment:
Same question as above.  It seems like we can do `Push` and `Pull` concurrently.